blobfuse/include/BlobStreamer.h

```diff
@@ -0,0 +1,191 @@
+#ifndef BLOBSTREAMER_H_
+#define BLOBSTREAMER_H_
+
+#include <blobfuse.h>
+#include <BlobfuseGlobals.h>
+#include<StorageBfsClientBase.h>
+
+#include <FileLockMap.h>
+#include <list>
+
+using namespace std;
+using namespace azure::storage_lite;
+using namespace azure::storage_adls;
+
+class CacheSizeCalculator
+{
+    public:
+        static CacheSizeCalculator* GetObj();
+
+        void SetMaxSize(uint64_t max) {
+            max_usage = max;
+        }   
+
+        void AddSize(uint64_t bytes) {
+            std::lock_guard<std::mutex> lock(m_mutex);
+            current_usage += bytes; 
+        }
+
+        void RemoveSize(uint64_t bytes) {
+            std::lock_guard<std::mutex> lock(m_mutex);
+            if (current_usage < bytes)
+                current_usage = 0;
```

Comment:
When will this case get hit? Would it be an error to hit this?

Reply:
Defensive coding. just in case we hit, we dont want to go in negative numbers.